src/server/js/routes/listing.js

```diff
@@ -18,21 +18,22 @@
  */
 
 import {Router} from 'express';
-import fbAdmin from '../services/firebase';
 const router = new Router();
 
 /* GET home page. */
 router.get('/', (req, res, next) => {
-  fbAdmin.database().ref('/products').once('value').then((snapshot) => {
-    res.render('listing', {
-      title: 'Holiday Pies — Pie Shop',
-      listing_name: 'Holiday Pies',
-      listing_description: 'Spice up your winter with our seasonal pies.',
-      products: snapshot.val(),
-      scripts: [
-        'https://www.gstatic.com/firebasejs/4.6.2/firebase.js',
-        'js/listing_main.js',
-      ],
+  import('../services/firebase').then((fbAdmin) => {
```

Comment:
Note: Moved this to a dynamic import to avoid the firebase initialization code running as soon as `listing.js` is loaded. This way it waits until the router gets a request before starting things off.